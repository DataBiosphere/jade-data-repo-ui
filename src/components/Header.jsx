import React from 'react';
import PropTypes from 'prop-types';
import rgba from 'polished/lib/color/rgba';
import Button from '@material-ui/core/Button';

import { appColor, headerHeight } from 'modules/theme';
import Logo from 'components/Logo';

const wrapperStyles = {
  backgroundColor: rgba(appColor, 0.9),
  height: headerHeight + 'px',
  left: 0,
  position: 'fixed',
  right: 0,
  top: 0,
  zIndex: 200,
};

const containerStyles = {
  alignItems: 'center',
  display: 'flex',
  flexWrap: 'wrap',
  height: '100%',
  justifyContent: 'space-between',
};

export default class Header extends React.PureComponent {
  static propTypes = {
    logOutClicked: PropTypes.func.isRequired,
    user: PropTypes.object.isRequired,
  };

  render() {
    const { user, logOutClicked } = this.props;
    return (
      <div>
        <div style={wrapperStyles}>
          <div style={containerStyles}>
            <Logo />
            {user.isAuthenticated && (
              <Button variant="outlined" onClick={logOutClicked}>
                Logout
              </Button>
            )}
          </div>
        </div>
      </div>
    );
  }
}
