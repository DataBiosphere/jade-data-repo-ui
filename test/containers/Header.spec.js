import React from 'react';

import Header from 'containers/Header';

const mockDispatch = jest.fn();

function setup() {
  const props = {
    dispatch: mockDispatch,
  };

  return mount(<Header {...props} />);
}

describe('Header', () => {
  const wrapper = setup();

  it('should render properly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
