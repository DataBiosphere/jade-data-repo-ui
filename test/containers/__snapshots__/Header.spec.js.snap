// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render properly 1`] = `
.c5 {
  display: inline-block;
  line-height: 0;
}

.c5 svg {
  height: auto;
  max-height: 100%;
  width: 16px;
}

.c2 {
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  height: 41px;
  width: 41px;
}

.c3 {
  bottom: 12px;
  color: #fff;
  font-size: 21px;
  left: 4px;
  position: relative;
}

.c0 {
  background-color: rgba(126,187,125,0.9);
  height: 70px;
  left: 0;
  position: fixed;
  right: 0;
  top: 0;
  z-index: 200;
}

.c0:before {
  background-color: #7ebb7d;
  bottom: 0;
  content: '';
  height: 0.2rem;
  left: 0;
  position: absolute;
  right: 0;
}

.c1 {
  box-sizing: border-box;
  margin-left: auto;
  margin-right: auto;
  padding-left: 16px;
  padding-right: 16px;
  max-width: 1440px;
  position: relative;
  width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  height: 100%;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding-bottom: 8px;
  padding-top: 8px;
}

.c4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #333;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1.3rem;
  padding: 8px;
}

.c4.active {
  color: #000;
}

.c4 span {
  display: inline-block;
  margin-right: 0.4rem;
  text-transform: uppercase;
}

@media (min-width:768px) {
  .c1 {
    padding-left: 32px;
    padding-right: 32px;
  }
}

@media (min-width:1024px) {
  .c4 {
    font-size: 1.6rem;
  }
}

<HeaderWrapper>
  <header
    className="c0"
  >
    <StyledContainer
      verticalPadding={false}
    >
      <div
        className="c1"
      >
        <HeaderLogo>
          <div>
            <Logo
              src={
                Object {
                  "ReactComponent": "SvgMock",
                }
              }
            >
              <img
                className="c2"
                src={
                  Object {
                    "ReactComponent": "SvgMock",
                  }
                }
              />
            </Logo>
            <Title>
              <span
                className="c3"
              >
                Jade Data Repository
              </span>
            </Title>
          </div>
        </HeaderLogo>
        <Logout
          onClick={[Function]}
        >
          <button
            className="c4"
            onClick={[Function]}
          >
            <span>
              logout
            </span>
            <IconWrapper
              color="#333"
              src="/media/icons/sign-out.svg"
              width={16}
            >
              <span
                className="isvg loading c5"
              />
            </IconWrapper>
          </button>
        </Logout>
      </div>
    </StyledContainer>
  </header>
</HeaderWrapper>
`;
